checker/cost.go

```diff
@@ -599,3 +605,30 @@ func (c *coster) newAstNode(e *exprpb.Expr) *astNode {
 	}
 	return &astNode{path: path, t: c.getType(e), expr: e, derivedSize: derivedSize}
 }
+
+func isScalar(t *exprpb.Type) bool {
+	switch kindOf(t) {
+	case kindPrimitive:
+		if t.GetPrimitive() != exprpb.Type_STRING && t.GetPrimitive() != exprpb.Type_BYTES {
+			return true
+		}
+	case kindWellKnown:
+		if t.GetWellKnown() == exprpb.Type_DURATION || t.GetWellKnown() == exprpb.Type_TIMESTAMP {
+			return true
+		}
+	case kindObject:
```

Comment:
`kindObject`/`MessageType` seems to be for directly-created protobufs (i.e, an expression containing something like `google.protobuf.Int32Value{}` or `google.protobuf.Duration{}`). Since those cases were handled already by [this part](https://github.com/google/cel-go/blob/master/checker/cost.go#L107-L110) of `ComputedSize` I've removed the `kindObject` case from `isScalar` entirely.